File: src/main/kotlin/xyz/mahmoudahmed/dsl/SourceBuilder.java
package xyz.mahmoudahmed.dsl;

public class SourceBuilder {
    private final String sourceName;
    private Format sourceFormat = Format.JSON; // Default format

    SourceBuilder(String sourceName) {
        this.sourceName = sourceName;
    }

    public SourceBuilder withFormat(Format format) {
        this.sourceFormat = format;
        return this;
    }

    public TargetBuilder to(String target) {
        return new TargetBuilder(sourceName, sourceFormat, target);
    }
}

File: src/main/kotlin/xyz/mahmoudahmed/dsl/builders/SourceBuilder.kt
package xyz.mahmoudahmed.dsl.builders

import xyz.mahmoudahmed.format.Format

data class SourceBuilder(
    private val sourceName: String,
    private val sourceFormat: Format = Format.JSON
) {
    fun withFormat(format: Format): SourceBuilder = copy(sourceFormat = format)

    fun to(target: String): TargetBuilder = TargetBuilder(sourceName, sourceFormat, target)
}
